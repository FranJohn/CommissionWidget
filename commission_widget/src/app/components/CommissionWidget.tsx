'use client'
import React, { ReactNode } from 'react';

/**
 * Props for the CommissionWidget component.
 */
interface CommissionWidgetInterface {
    children: ReactNode;
}

/**
 * CommissionWidget component for handling commission calculations.
 * @param {CommissionWidgetProps} props - The props for CommissionWidget.
 * @returns {JSX.Element} JSX for CommissionWidget component.
 */
const CommissionWidget: React.FC<CommissionWidgetInterface> = ({ children }) => {
    return (
        <div className="widget-container">
            {children}
        </div>
    );
};

export default CommissionWidget;
