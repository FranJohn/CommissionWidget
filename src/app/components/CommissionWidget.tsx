'use client'
import React, { ReactNode } from 'react';

/**
 * Props for the CommissionWidget component.
 */
interface CommissionWidgetInterface {
    children: ReactNode;
    title: string;
}

/**
 * CommissionWidget component for handling commission calculations.
 * @param {CommissionWidgetProps} props - The props for CommissionWidget.
 * @returns {JSX.Element} JSX for CommissionWidget component.
 */
const CommissionWidget: React.FC<CommissionWidgetInterface> = ({ children, title }) => {
    return (
        <div className="widget-title-container">
            <h2>{title}</h2>
            <div className="widget-container">
                {children}
            </div>
        </div>
    );
};

export default CommissionWidget;
