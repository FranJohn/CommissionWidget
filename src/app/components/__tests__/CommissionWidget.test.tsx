import '@testing-library/jest-dom';
import React from 'react';
import { render } from '@testing-library/react';
import CommissionWidget from '../CommissionWidget';

describe('CommissionWidget Component', () => {
  it('renders with the provided title', () => {
    const title = 'Test Title';
    const { getByText } = render(<CommissionWidget title={title}>Child content</CommissionWidget>);
    expect(getByText(title)).toBeInTheDocument();
  });

  it('renders children content', () => {
    const childContent = 'Child content';
    const { getByText } = render(<CommissionWidget title="Test Title">{childContent}</CommissionWidget>);
    expect(getByText(childContent)).toBeInTheDocument();
  });
});
