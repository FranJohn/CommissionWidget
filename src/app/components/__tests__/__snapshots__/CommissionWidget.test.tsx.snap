// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommissionWidget Component matches the CommissionWidget snapshot 1`] = `
<div>
  <div
    class="widget-title-container"
  >
    <h2>
      Test Title
    </h2>
    <div
      class="widget-container"
    >
      Child content
    </div>
  </div>
</div>
`;
